src/cirrus/management/deployment_pointer.py

```diff
@@ -1,164 +1,161 @@
-    def get(
+    def get_deployment_by_name(
         cls,
         deployment_name: str,
-        deployment_prefix: str = DEFAULT_CIRRUS_DEPLOYMENT_PREFIX,
+        deployment_prefix: str = PARAMETER_PREFIX,
         region: str | None = None,
         session: boto3.Session | None = None,
     ):
-        ssm = get_client("ssm", region=region, session=session)
-        return cls._from_parameter(
-            ssm.get_parameter(
-                Name=f"{deployment_prefix}{deployment_name}",
-            )["Parameter"],
-            prefix=deployment_prefix,
+        """
+        Retrieve a single deployment configuration by name from parameter store
+        """
+        deployment = cls._get_deployments(
+            deployment_prefix,
+            deployment_name,
+            region,
+            session,
         )
+        if deployment:
+            return deployment[0]
+        return None
 
     @classmethod
-    def list(
+    def list_deployments(
         cls,
-        deployment_prefix: str = DEFAULT_CIRRUS_DEPLOYMENT_PREFIX,
+        deployment_prefix: str = PARAMETER_PREFIX,
         region: str | None = None,
         session: boto3.Session | None = None,
     ) -> Iterator[DeploymentPointer]:
-        ssm = get_client("ssm", region=region, session=session)
-        resp = ssm.get_parameters_by_path(Path=deployment_prefix)
-        for param in resp["Parameters"]:
-            yield cls._from_parameter(
-                param,
-                prefix=deployment_prefix,
-            )
+        """Retrieve and list all deployments in parameter store"""
+        deployments = cls._get_deployments(
+            deployment_prefix,
+            region=region,
+            session=session,
+        )
+        if deployments:
+            yield from deployments
+        return None
 
-    def get_config(
-        self,
-        session: boto3.Session | None = None,
-    ) -> dict[str, Any]:
-        return json.loads(self.pointer.resolve().fetch(session=session))
+    @classmethod
+    def parsed_parameter(cls, param: dict, prefix: str) -> tuple[str, str, str]:
+        """split param into name, value, and deployment name components"""
+        return (
+            param["Name"].split("/")[-1],
+            param["Value"],
+            param["Name"][len(prefix) : param["Name"].rindex("/")],
+        )
+
+    @classmethod
+    def cirrus_components(
+        cls,
+        parameters: list[tuple[str, str, str]],
+        name: str,
+    ) -> dict[str, str]:
+        return {
+            param_name: value
+            for param_name, value, deployment_name in parameters
+            if deployment_name == name
+        }
+
+    @classmethod
+    def validate_vars(cls, parameters: list[tuple[str, str, str]]):
+        parameter_names = [name for name, _, _ in parameters]
+        for field in fields(DeploymentEnvVars):
+            if field.name not in parameter_names:
+                raise MissingParameterError(field.name)
```

Comment:
Rather than using an iterative approach like this, I would offer the option of using sets. Instead of having a list of required fields like I suggested, make that a set of required fields. Then here, convert your parameter_names into a set, and subtract: `missing_fields = required_fields - provided_fields`.

In addition to this suggestion, I have the two questions:
* Why is `parameters` not a dict?
* Why is this a `classmethod` if `cls` is not used?